Task2/MyThreadLazy.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+
+class MyThreadLazy<T> : ILazy<T>
+{
+    private Func<T> supplier;
+    private T value;
+    private bool isCreatedValue = false;
```

Comment:
```suggestion
    private volatile bool isCreatedValue = false;
```
эта переменная должна быть volatile, потому что операция записи не атомарная => не факт, что внутри lock операция записи не будет прервана. Тогда другой поток войдёт в функцию, увидит, что `isCreatedValue` ложно, и выполнит вычисление ещё раз. А нам этого как раз не надо